beancount/parser/booking_method.py

```diff
@@ -170,77 +168,82 @@ def booking_method_NONE(entry, posting, matches):
     return [posting], [], False
 
 
+def rebook_inventory_at_average_cost(full_inventory: Inventory,
+                                     account,
+                                     unit_currency,
+                                     cost_currency):
+    """ Returns a set of postings that rebook an inventory at average cost.
+
+    An inventory can be held in multiple cost currencies, so we treat units
+    held in differing cost currencies completely separately.
+
+    Args:
+        full_inventory: The ante-inventory that we want to rebook at average cost.
+          The inventory passed in will be updated.
+        account: The account name this inventory is held for.
+        unit_currency: The unit currency (commodity) that we want to rebook our
+          holdings for.
+        cost_currency: The cost currency to filter the inventory's positions,
+          since we treat units held in differing cost currencies completely
+          indepdendently.
+
+    Returns: a list of postings that will negate the inventory and rebook it
+    as an average position. If the inventory is already a single position for
+    the specified cost_currency, an empty list is returned as no rebooking is
+    necessary.
+    """
+    filtered_inv = inventory.Inventory()
+    for inv_position in full_inventory:
+        if inv_position.units.currency == unit_currency and \
+        inv_position.cost.currency == cost_currency:
+            filtered_inv.add_position(inv_position)
+
+    if len(filtered_inv) <= 1:
+        return []
+
+    booked_postings = []
+    avg_position = filtered_inv.average().get_only_position()
+    # Negate existing inventory.
+    for inv_position in filtered_inv:
+        full_inventory.add_amount(-inv_position.units, inv_position.cost)
```

Comment:
Avoid this and the next update, and let the callers update their inventories.
(Avoid the subtle mutation.)

Reply:
Done